modules/ppcp-settings/resources/js/Components/ReusableComponents/ConnectionInfo.js

```diff
@@ -1,54 +1,58 @@
 import { __ } from '@wordpress/i18n';
-import { useState } from '@wordpress/element';
+import { useState, useEffect } from '@wordpress/element';
+import {CommonHooks} from "../../data";
 
-const ConnectionInfo = ( { connectionStatusDataDefault } ) => {
-	const [ connectionData, setConnectionData ] = useState( {
-		...connectionStatusDataDefault,
-	} );
 
-	const toggleStatusClassName = [ 'ppcp-r-connection-status__status-toggle' ];
+const ConnectionInfo = () => {
+    const { merchant } = CommonHooks.useMerchantInfo();
+    const [connectionData, setConnectionData] = useState(getDefaultConnectionStatusData(merchant));
 
-	if ( connectionData.showAllData ) {
-		toggleStatusClassName.push(
-			'ppcp-r-connection-status__status-toggle--toggled'
-		);
-	}
+    useEffect(() => {
+        if (merchant) {
+            setConnectionData(getDefaultConnectionStatusData(merchant));
+        }
+    }, [merchant]);
 
 	return (
-		<div className="ppcp-r-connection-status__data">
-			<div className="ppcp-r-connection-status__status-row ppcp-r-connection-status__status-row--first">
-				<span className="ppcp-r-connection-status__status-label">
-					{ __( 'Merchant ID', 'woocommerce-paypal-payments' ) }
-				</span>
-				<span className="ppcp-r-connection-status__status-value">
-					{ connectionData.merchantId }
-				</span>
-			</div>
-			<div className="ppcp-r-connection-status__status-row">
-				<span className="ppcp-r-connection-status__status-label">
-					{ __( 'Email address', 'woocommerce-paypal-payments' ) }
-				</span>
-				<span className="ppcp-r-connection-status__status-value">
-					{ connectionData.email }
-				</span>
-			</div>
-			<div className="ppcp-r-connection-status__status-row">
-				<span className="ppcp-r-connection-status__status-label">
-					{ __( 'Client ID', 'woocommerce-paypal-payments' ) }
-				</span>
-				<span className="ppcp-r-connection-status__status-value">
-					{ connectionData.clientId }
-				</span>
-			</div>
-		</div>
-	);
+        <div className="ppcp-r-connection-status__data">
+            {renderStatusRow(
+                __('Merchant ID', 'woocommerce-paypal-payments'),
+                connectionData.merchantId
+            )}
+            {renderStatusRow(
+                __('Email address', 'woocommerce-paypal-payments'),
+                connectionData.email
+            )}
+            {renderStatusRow(
+                __('Client ID', 'woocommerce-paypal-payments'),
+                connectionData.clientId
+            )}
+        </div>
+    );
 };
 export default ConnectionInfo;
+export const getDefaultConnectionStatusData = (merchant = null) => {
```

Comment:
This method just returns merchant data with renamed properties - let's completely remove this method, as the `merchant` object returned by the `useMerchant()` already contains all needed details